substrate/frame/core-fellowship/src/lib.rs

```diff
@@ -337,8 +338,22 @@ pub mod pallet {
 		#[pallet::call_index(1)]
 		pub fn set_params(origin: OriginFor<T>, params: Box<ParamsOf<T, I>>) -> DispatchResult {
 			T::ParamsOrigin::ensure_origin_or_root(origin)?;
+			// Retrieves the current max rank allowed.
+			let max_rank = T::Members::max_rank();
+
+			// Validate the lengths of the vectors in params are equal to max_rank.
+			let expected_length = max_rank as usize; // Convert max_rank to usize for comparison.
+			ensure!(
+				params.active_salary.len() == expected_length &&
+					params.passive_salary.len() == expected_length &&
```

Comment:
If you use bounded vec, then there is no issue here either.

Reply:
Thanks for the review!